resources/images/text.go

```diff
@@ -77,33 +78,67 @@ func (f textFilter) Draw(dst draw.Image, src image.Image, options *gift.Options)
 
 	gift.New().Draw(dst, src)
 
-	// Draw text, consider and include linebreaks
 	maxWidth := dst.Bounds().Dx() - 20
-	fontHeight := face.Metrics().Ascent.Ceil()
 
-	// Correct y position based on font and size
-	f.y = f.y + fontHeight
+	// Default left alignment
+	availableWidth := maxWidth - f.x
+	if f.alignx == "center" {
+		availableWidth = min((maxWidth-f.x), f.x) * 2
+	} else if f.alignx == "right" {
+		availableWidth = f.x
+	}
 
-	// Start position
-	y := f.y
-	d.Dot = fixed.P(f.x, f.y)
+	fontHeight := face.Metrics().Ascent.Ceil()
 
-	// Draw text line by line, breaking each line at the maximum width.
+	// Calculate lines, consider and include linebreaks
+	finalLines := []string{}
 	f.text = strings.ReplaceAll(f.text, "\r", "")
 	for _, line := range strings.Split(f.text, "\n") {
+		currentLine := ""
+		// Break each line at the maximum width.
 		for _, str := range strings.Fields(line) {
-			strWidth := font.MeasureString(face, str)
-			if (d.Dot.X.Ceil() + strWidth.Ceil()) >= maxWidth {
-				y = y + fontHeight + f.linespacing
-				d.Dot = fixed.P(f.x, y)
+			fieldStrWidth := font.MeasureString(face, str)
+			currentLineStrWidth := font.MeasureString(face, currentLine)
+
+			if (currentLineStrWidth.Ceil() + fieldStrWidth.Ceil()) >= availableWidth {
+				finalLines = append(finalLines, currentLine)
+				currentLine = ""
 			}
-			d.DrawString(str + " ")
+			currentLine += str + " "
 		}
+		finalLines = append(finalLines, currentLine)
+	}
+
+	// Correct y position based on font and size
+	f.y = f.y + fontHeight
+
+	// Start position
+	y := f.y
+
+	// Draw text line by line
+	for _, line := range finalLines {
+		line = strings.TrimSpace(line)
+		strWidth := font.MeasureString(face, line)
+		x := calcX(f.alignx, f.x, strWidth)
+		d.Dot = fixed.P(x, y)
+		d.DrawString(line)
 		y = y + fontHeight + f.linespacing
-		d.Dot = fixed.P(f.x, y)
 	}
 }
 
 func (f textFilter) Bounds(srcBounds image.Rectangle) image.Rectangle {
 	return image.Rect(0, 0, srcBounds.Dx(), srcBounds.Dy())
 }
+
+func calcX(alignX string, x int, strWidth fixed.Int26_6) (rightX int) {
```

Comment:
Named return values in Go is, I think, in general frowned upon, and only used if you really need it (the one case I can come up with is when you need to change the return value in defer/recover).

Also, if `calcX` isn't in general useful, I would just pull it into the method, e.g.: `calcX := func(alignX string, x int, strWidth fixed.Int26_6) (rightX int) {...}`